pkg/action/failback.go

```diff
@@ -274,90 +287,9 @@ func (ac *ActionController) performLastMileMigration(action *storkv1.Action) {
-			return
-		}
-	} else if action.Spec.ActionType == storkv1.ActionTypeFailback {
-		// get destination i.e. current cluster's config
-		config = ac.config
-	}
-
-	storkClient, err := storkops.NewForConfig(config)
-	if err != nil {
-		msg := fmt.Sprintf("Failed to create the stork client to access cluster with the config %v", config.Host)
-		log.ActionLog(action).Errorf(msg)
-		action.Status.Status = storkv1.ActionStatusFailed
-		action.Status.Reason = msg
-		ac.recorder.Event(action,
-			v1.EventTypeWarning,
-			string(storkv1.ActionStatusFailed),
-			msg)
-		ac.updateAction(action)
-		return
-	}
-
-	migrationName := getLastMileMigrationName(migrationSchedule.Name, string(action.Spec.ActionType), utils.GetShortUID(string(action.UID)))
-	migrationObject, err := storkClient.GetMigration(migrationName, migrationSchedule.Namespace)
-	if err != nil {
-		if k8sErrors.IsNotFound(err) {
-			// If last mile migration CR does not exist, create one
-			migration := getLastMileMigrationSpec(migrationSchedule, string(action.Spec.ActionType), utils.GetShortUID(string(action.UID)))
-			_, err := storkClient.CreateMigration(migration)
-			if err != nil {
-				msg := fmt.Sprintf("Creating last mile migration from the MigrationSchedule %s failed: %v", migrationSchedule.GetName(), err)
-				ac.recorder.Event(action,
-					v1.EventTypeWarning,
-					string(storkv1.ActionStatusFailed),
-					msg)
-				ac.updateAction(action)
-				return
-			}
-			action.Status.Status = storkv1.ActionStatusInProgress
-			ac.updateAction(action)
-			return
-		} else {
-			msg := fmt.Sprintf("Failed to get the last mile migration object %s: %v", migrationName, err)
-			log.ActionLog(action).Errorf(msg)
-			ac.recorder.Event(action,
-				v1.EventTypeWarning,
-				string(storkv1.ActionStatusFailed),
-				msg)
-			ac.updateAction(action)
-			return
-		}
-	}
-
-	if migrationObject.Status.Stage == storkv1.MigrationStageFinal {
-		if migrationObject.Status.Status == storkv1.MigrationStatusSuccessful || migrationObject.Status.Status == storkv1.MigrationStatusPartialSuccess {
-			action.Status.Status = storkv1.ActionStatusSuccessful
-		} else {
-			msg := fmt.Sprintf("Failing %s operation as the last mile migration %s failed: status %s", action.Spec.ActionType, migrationObject.Name, migrationObject.Status.Status)
-			log.ActionLog(action).Errorf(msg)
-			action.Status.Status = storkv1.ActionStatusFailed
-			action.Status.Reason = msg
-			ac.recorder.Event(action,
-				v1.EventTypeWarning,
-				string(storkv1.ActionStatusFailed),
-				msg)
-		}
-		ac.updateAction(action)
-		return
-	}
-	log.ActionLog(action).Infof("Last mile migration %s is still running, currently in stage: %s", migrationObject.Name, migrationObject.Status.Stage)
+	// In failback last-mile-migration is created in destination->source direction
+	config := ac.config
+	ac.createLastMileMigration(action, config, migrationSchedule)
 }
 
 // getLastMileMigrationSpec will get a migration spec from the given migrationschedule spec
```

Comment:
I will move the other methods common to both failover/failback such as activateClusterDuringDR, deactivateClusterDuringDR, getLastMileMigratioonSpec etc. to the drutils file in another PR.
Didn't do that now to make review easier